react/week3/todolist/src/components/TodoList.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+
+export default class TodoList extends Component {
+   render() {
+      const { isLoading, error } = this.props;
+      return (
+         <>
+            {error ? <p>{error.message}</p> : null}
+            {!isLoading ? (
```

Comment:
I love the way you've structured this to handle errors and loading! 

If you wanted to take this a step further to improve readability, I would remove the error and loading from the main render function and put them in their own condition like: 

```jsx
render() {
      const { isLoading, error } = this.props;
      
      if (error) {
            return <p>{error.message}</p>
      } 

      if (loading) {
            return <h3>Loading...</h3>
      }

      return (
// return the actual component here
      )
```

It's definitely more verbose to write it this way, however I think it makes it super clear to read if someone else needs to read your code. Nesting ternaries in components is definitely common in react, however arguably a bit of an anti-pattern regarding readability in some cases.